connectors/sources/zoom.py

```diff
@@ -0,0 +1,540 @@
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+        return True
+
+    async def _get_document_with_content(self, doc):
+        document = {
+            "_id": doc["id"],
+            "_timestamp": doc["date_time"],
+        }
+
+        temp_filename = ""
+
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+            response = await self.client.get_file_content(
+                download_url=doc["download_url"]
+            )
+            await async_buffer.write(response.encode("utf-8"))
+            temp_filename = str(async_buffer.name)
+
+        await asyncio.to_thread(
+            convert_to_b64,
+            source=temp_filename,
+        )
+        async with aiofiles.open(file=temp_filename, mode="r") as target_file:
+            # base64 on macOS will add a EOL, so we strip() here
+            document["_attachment"] = (await target_file.read()).strip()
+        try:
+            await remove(temp_filename)
+        except Exception as exception:
+            self._logger.warning(
+                f"Could not remove file: {temp_filename}. Error: {exception}"
+            )
+
+        return document
+
+    async def get_content(self, doc, timestamp=None, doit=False):
+        attachment_size = doc["file_size"]
+        if not (doit and attachment_size > 0):
+            return
+
+        attachment_name = doc["file_name"]
+
+        attachment_extension = (
+            attachment_name[attachment_name.rfind(".") :]  # noqa
+            if "." in attachment_name
+            else ""
+        )
+
+        if not self._pre_checks_for_get_content(
+            attachment_extension=attachment_extension,
+            attachment_name=attachment_name,
+            attachment_size=attachment_size,
+        ):
+            return
+
+        self._logger.debug(f"Downloading {attachment_name}")
+        return await self._get_document_with_content(doc)
+
+    async def fetch_previous_meeting_details(self, meeting_id):
+        previous_meeting = await self.client.get_past_meeting(meeting_id=meeting_id)
+
+        if not previous_meeting:
+            return
+
+        participants = []
+        async for participant in self.client.get_past_meeting_participants(
+            meeting_id=meeting_id
+        ):
+            participants.append(participant)
+
+        previous_meeting["participants"] = participants
+        return previous_meeting
+
+    async def fetch_recordings(self, user_id):
+        async for recording in self.client.get_recordings(user_id=user_id):
+            yield recording
+
+        async for recording in self.client.get_trash_recordings(user_id=user_id):
+            yield recording
+
+    async def get_docs(self, filtering=None):
+        async for user in self.client.get_users():
+            yield self._format_doc(doc=user, doc_time=iso_utc()), None
```

Comment:
are we sure we always want to sync user records? I expect that the most value for a zoom connector comes from the chat and from the transcript. Should customers be able to opt out of syncing users @danajuratoni ?

Reply:
I assume user records are the meeting's attendees. I see use cases where search users would want to filter / search by attendees. So we should include this information, making it configurable would be a nice flexibility improvement. Are there any performance concerns if we add them by default?

Reply:
Participants field references only display names, let's ingest users to also have [better feature similarity with the WS](https://github.com/elastic/enterprise-search-zoom-connector#data-extraction-and-syncing) connector

Reply:
that's great @abhishekjoshi-crest, we should ingest the participants info returned by the linked API